File: src/main/java/no/nav/service/pensjon/regler/to/BeregnYtelseRequest.java
package no.nav.service.pensjon.regler.to;

import java.util.Date;
import java.util.Vector;

import no.nav.domain.pensjon.regler.grunnlag.InfoPavirkendeYtelse;
import no.nav.domain.pensjon.regler.krav.Kravhode;
import no.nav.domain.pensjon.regler.vedtak.VilkarsVedtak;

public class BeregnYtelseRequest extends ServiceRequest {


	private Kravhode kravhode;

	private Vector<VilkarsVedtak> vilkarsvedtaksliste = new Vector<VilkarsVedtak>();

	private Date fom;

	private Date tom;

	private boolean ektefelleMottarPensjon;

	private boolean beregnForsorgingstillegg;

	private boolean beregnInstitusjonsopphold;
	
	private InfoPavirkendeYtelse infoPavirkendeYtelse;

	/**
	 * Ikke brukt i PREG
	 */
	private Date vedtakVirkFom;


}

File: src/main/kotlin/no/nav/pensjon/regler/to/BeregnYtelseRequest.kt
package no.nav.pensjon.regler.to

import no.nav.pensjon.regler.domain.grunnlag.InfoPavirkendeYtelse
import no.nav.pensjon.regler.domain.krav.Kravhode
import no.nav.pensjon.regler.domain.vedtak.VilkarsVedtak
import java.util.*
import java.time.LocalDate

class BeregnYtelseRequest : ServiceRequest() {
    var kravhode: Kravhode? = null
    var vilkarsvedtaksliste = Vector<VilkarsVedtak>()
    var fomLd: LocalDate? = null
    var tomLd: LocalDate? = null
    var ektefelleMottarPensjon = false
    var beregnForsorgingstillegg = false
    var beregnInstitusjonsopphold = false
    var infoPavirkendeYtelse: InfoPavirkendeYtelse? = null
    var vedtakVirkFomLd: LocalDate? = null
}